core/impl/src/main/java/com/blazebit/persistence/impl/function/window/AbstractWindowFunction.java

```diff
@@ -0,0 +1,557 @@
+                                    currentFrame.setType(FramePositionType.UNBOUNDED_FOLLOWING);
+                                    break;
+                                default:
+                                    currentFrame.setExpression(argument);
+                                    break;
+                            }
+
+                            break;
+                        default:
+                            throw new IllegalArgumentException("No branch for " + mode);
+                    }
+            }
+        }
+
+        render(context, windowFunction);
+    }
+
+    protected abstract String getFunctionName();
+
+
+    private static Order getOrder(String sort, String expression) {
+        String type = sort.trim().toUpperCase();
+
+        if ("'ASC'".equals(type)) {
+            return new Order(expression, true, null);
+        } else if ("'DESC'".equals(type)) {
+            return new Order(expression, false, null);
+        } else if ("'ASC NULLS FIRST'".equals(type)) {
+            return new Order(expression, true, true);
+        } else if ("'ASC NULLS LAST'".equals(type)) {
+            return new Order(expression, true, false);
+        } else if ("'DESC NULLS FIRST'".equals(type)) {
+            return new Order(expression, false, true);
+        } else if ("'DESC NULLS LAST'".equals(type)) {
+            return new Order(expression, false, false);
+        }
+
+        return null;
+    }
+    
+    protected void render(FunctionRenderContext context, WindowFunction windowFunction) {
+        renderFunction(context, windowFunction);
+
+        List<String> filterExpressions = windowFunction.getFilterExpressions();
+        if (filterExpressions != null && !filterExpressions.isEmpty()) {
+            renderFilterExpressions(context, filterExpressions);
+        }
+
+        context.addChunk("OVER (");
+
+        WindowDefinition window = windowFunction.getWindow();
+
+        Partition partition = window.getPartition();
+        if (partition != null && partition.getExpressions() != null && ! partition.getExpressions().isEmpty()) {
+            render(context, partition);
+        }
+
+        List<Order> orderBys = window.getOrderBys();
+        if (orderBys != null && ! orderBys.isEmpty()) {
+            render(context, orderBys);
+        }
+
+        Frame frame = window.getFrame();
+        if (frame != null) {
+            render(context, frame);
+
+        }
+
+        context.addChunk(")");
+    }
+
+    protected void renderFunction(FunctionRenderContext context, WindowFunction windowFunction) {
+        context.addChunk(windowFunction.getFunctionName());
+        context.addChunk("(");
+
+        renderArguments(context, windowFunction);
+
+        context.addChunk(") ");
+    }
+
+    protected void renderArguments(FunctionRenderContext context, WindowFunction windowFunction) {
+        Iterator<String> argumentIterator = windowFunction.getArguments().iterator();
```

Comment:
Since we know we are dealing with array lists here, I'd prefer an iteration like:

```
        List<String> arguments = windowFunction.getArguments();
        int size = arguments.size();
        if (size != 0) {
            context.addChunk(arguments.get(0));
            for (int i = 1; i < size; i++) {
                context.addChunk(", ");
                context.addChunk(arguments.get(i));
            }
        }
```

Also for all other iterations in this class